Real data v2 home

0xjocke opened the pull request:
- Fixes up some inconsistencies with network id. 
- Use container pattern for v2-components
- Rename Banner to CRatioBanner and make the text dynamic
- Include `liquidationDeadlineForAccount` in debt data (this is how we know if someone is flagged)

There still some todo's in here, but might be worth getting some feedback on the architecture. Happy to continue working on this branch or create a new PR resolving the todos.

Also build is failing due to some problems with types. When it's fixed and #331  is merged I'll rebase

0xjocke committed:
Lets use a number for network id.
- Also our hooks needs to handle null/ undefined for network/ provider since that's what the connector returns.

0xjocke committed:
Use the container pattern for our v2 components

0xjocke committed:
Rename Banner to CRatioBanner and make the text dynamic based on data

0xjocke committed:
Fix deps

0xjocke committed:
liquidationDeadlineForAccount to be included in debtData
- This lets us know if a user is flagged

jmzwar commented on v2/components/BalanceBox/BalanceBox.tsx:
```diff
@@ -58,3 +60,22 @@ export const BalanceBox = ({
+  provider: SynthetixProvider | null;
+  walletAddress: string | null;
+};
+export const BalanceBox: React.FC<Props> = ({ networkId, provider, walletAddress }) => {
```
Is the intention to keep the wrapper component in the same file as the UI component?

bachstatter replied:
yep

jmzwar commented on v2/components/CRatioBanner/CRatioBanner.tsx:
```diff
@@ -0,0 +1,70 @@
+}) => {
+  if (variant === 'success') {
+    return <Center bgGradient={theme.gradients['green-cyan'][500]}>{children}</Center>;
+  }
```
Would it be possible to just return the bgGradient or bgColor here rather than the wrapper component? And then call the function in the props of the actual wrapper?

bachstatter replied:
yea sure

jmzwar reviewed (commented):
Looks good. Just the two questions.

noisekit reviewed (approved):
I really like how the ui/container component separation works out 💪

bachstatter commented on v2/components/BalanceBox/BalanceBox.tsx:
```diff
@@ -3,15 +3,17 @@ import { Box, Center, Flex, Text, Tooltip, Progress } from '@chakra-ui/react';
   transferable,
   stakedSnx,
-}: PropsWithChildren<Props>) => {
+}: PropsWithChildren<UiProps>) => {
```
Yea I'm not fussed about this.. I think I would do inline unless there's quite a few props normally. Can change to inline

bachstatter commented on v2/components/CRatioHealthCard/CRatioHealthCard.tsx:
```diff
@@ -40,7 +42,7 @@ export const CRatioHealthCard: React.FC<Props> = ({
             fontFamily="mono"
           >
-            {currentCRatioPercentage}%
+            {Math.round(currentCRatioPercentage)}%
```
hmm yea maybe.. did consider this. I think the contracts have a little bit on a buffer for ratio's so probably doesn't matter too much

bachstatter commented on v2/lib/useSynthetixContracts/useSynthetixContracts.ts:
```diff
@@ -18,29 +18,45 @@ export const NetworkNameById = {
 
   return useQuery(
-    [contractNames, networkName, networkId],
+    [contractNames, networkId],
```
hmm yea. I did consider it, but thought it would be quite common to grab a few contracts in the hooks. `useDebtData` uses 3. And it gets a little uglier to have three react-queries added to enabled.. But if types just works its probably worth it

bachstatter replied:
types is working out so moving to this structure

0xjocke committed:
Create a hook for each contract.

0xjocke committed:
Remove contracts file, each hook can just specify it's contracts

bachstatter commented on v2/lib/useSynthetixContracts/useSystemSettings.ts:
```diff
@@ -0,0 +1,34 @@
+    throw Error(`${networkId} is not supported`);
+  }
+  const networkName = NetworkNameById[networkId];
+  const { address, abi } = await contracts['SystemSettings'][networkName]();
```
yep def, fixed

0xjocke committed:
Use floor instead

0xjocke committed:
Deps fix

0xjocke committed:
PR feedback just return styles for container

0xjocke committed:
Lint, dont know how that got in there

0xjocke committed:
Percentage is already multiplied with 100

0xjocke committed:
Fix some issues in debt data
- Fetch DeadlineForAccount
- correct enabled condition
- correct type for processQueryData

0xjocke committed:
Add enable condition and staleTime for contract queries

0xjocke committed:
Update v2/lib/useDebtData/useDebtData.ts

Co-authored-by: nk <nikita@cc.snxdao.io>

bachstatter commented on v2/ui/content/V2Home.tsx:
```diff
@@ -1,49 +1,39 @@
-            currentCRatioPercentage={currentCRatioPercentage}
-            liquidationCratioPercentage={liquidationCratioPercentage}
-          />
+          <CRatioHealthCard {...propsFromConnector} />
+          <MainActionCards {...propsFromConnector} />
```
We still dont have a context to get network id and the provider from the Connector container. So for now I use props for that

bachstatter replied:
Using context for this now

bachstatter commented on v2/ui/content/V2Home.tsx:
```diff
@@ -1,49 +1,39 @@
     <Box>
       <Box height="42px" position="absolute" left="0" right="0">
-        <Banner variant={variant} text="You can collect your weekly rewards" countDown={epoch} />
+        <CRatioBanner {...propsFromConnector} />
```
It's quite an effort to move the connector to a separate package

bachstatter replied:
Using a context for this now

0xjocke committed:
Add ContractContext

jmzwar committed:
unset max width

0xjocke committed:
Wallet address should be in context too

0xjocke committed:
Add dotenv

0xjocke committed:
Rely on context rather than passing arguments

bachstatter commented on v2/ui/App.tsx:
```diff
@@ -62,11 +64,20 @@ function InnerApp() {
-        </Layout>
+        <ContractContext.Provider
+          value={{
+            networkId: isSupportedNetworkId(networkId) ? networkId : 1,
+            provider: provider || L1DefaultProvider,
+            signer: signer || null,
+            walletAddress,
```
Lets have a chat about this. The idea to have a default provider and a default network is that we can fetch data before the user is connected.  But yea we could just instantiate a default provider/network in each contract hook if we want to keep the context small

0xjocke committed:
Remove provider and signer from Contract context

0xjocke committed:
Add signer context

0xjocke committed:
Use signer context

0xjocke committed:
Remove unnecessary props

0xjocke committed:
Deps

noisekit reviewed (approved):
Love this very much

bachstatter commented on v2/ui/content/V2Home.tsx:
```diff
@@ -1,52 +1,31 @@
-            currentCRatioPercentage={currentCRatioPercentage}
-            liquidationCratioPercentage={liquidationCratioPercentage}
-          />
+          <CRatioHealthCard />
+          <MainActionCards />
```
😌
